testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/asynch/AsyncPostProcessingTest.java

```diff
@@ -46,15 +46,15 @@
         war.addClasses(TestUtil.class, PortProviderUtil.class);
         war.addAsWebInfResource(AsyncPostProcessingTest.class.getPackage(), "AsyncPostProcessingTestWeb.xml", "web.xml");
         // Arquillian in the deployment
-        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(new ReflectPermission("suppressAccessChecks"),
-                new LoggingPermission("control", ""),
+        war.addAsManifestResource(PermissionUtil.createPermissionsXmlAsset(
+                new ReflectPermission("suppressAccessChecks"),
                 new PropertyPermission("arquillian.*", "read"),
                 new PropertyPermission("ipv6", "read"),
                 new PropertyPermission("node", "read"),
                 new PropertyPermission("org.jboss.resteasy.port", "read"),
                 new RuntimePermission("accessDeclaredMembers"),
                 new RuntimePermission("getenv.RESTEASY_PORT"),
-                new SecurityPermission("insertProvider"),
+                new RuntimePermission("getClassLoader"),
```

Comment:
Just as a confirmation, is the getClassloader permission needed here and everywhere else in this PR because of jboss-modules? or maybe RESTEasy is actually needing it?

Reply:
Just tried, this is the stacktrace: https://paste.fedoraproject.org/paste/bg8Xppg3l1-nrP6TdSRP8Q/raw
So javax.ws.rs.ext.FactoryFinder.getModuleClassLoader method is calling getClassloader through reflection and that's failing because WildFlySecurityManager checks for the permissions before allowing that.
If we want to avoid asking users to provide the permission here (and I think we should, it's really a detail out of user control), we should likely fix our jboss-jaxrs-api project: https://github.com/jboss/jboss-jaxrs-api_spec/blob/master/src/main/java/javax/ws/rs/ext/FactoryFinder.java#L255 .

This is basically the same that has been done in the past for jaxws at https://github.com/jboss/jboss-jaxws-api_spec/commit/c1eb89a088255f071f01bbb7561c77cb3052239d#diff-37f7e6b2d322553ab40f0cc7f1290683L178 (see https://issues.jboss.org/browse/JBEE-176 ).

Reply:
Making the code change in https://github.com/jboss/jboss-jaxrs-api_spec FactoryFinder
resolved the "getClassLoader" issue there but a new "getClassLoader" issue is now
present.  See  https://paste.fedoraproject.org/paste/86J2ZO4cXvWaHKjjU75YkQ 

The call in org.jboss.resteasy.core.ContextParameterInjector.createProxy(ContextParameterInjector.java:168)
is to java.lang.reflect.Proxy which does perform a SecurityManager check. 

Suggestions for addressing this one?

Reply:
Judging from the stacktrace, the call resulting in the getClassloader permission check being performed is the "rawType.getClassLoader()" one that's used to compute the first parameter for the proxy creation. So I think you could use a privileged action to get that classloader, right?